aredis/client.py

```diff
@@ -160,6 +157,7 @@ def set_response_callback(self, command, callback):
         except CancelledError:
             # do not retry when coroutine is cancelled
             connection.disconnect()
+            raise
```

Comment:
why should a CancelledError be raised?

Reply:
If a user of aredis wants to do cleanup when a task is cancelled, or log a particular message, it will need to know about that exception. 

I've done some 'googling' and some library re-raise CancelledError (basically a library will do its own cleanup, like yourself where you call disconnect(), then re-raise the exception.

https://www.programcreek.com/python/example/81224/asyncio.CancelledError

For error logging, a user might want to print a special message if a connection is dropped, but no message if a cancellation is requested explicitely (such as when a system is rebooted etc...).